snapshot/doc/user-guide.md

```diff
@@ -0,0 +1,112 @@
+
+### Persistent Volume Claim and Persistent Volume
+The user already created a Persistent Volume Claim that is bound to a Persistent Volume. The Persistent Volume type must be one of the snaphot supported Persistent Volume types.
+
+### Snapshot Promoter
+An admin created a Storage Class like the one shown below:
+```yaml
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: snapshot-promoter
+provisioner: volumesnapshot.external-storage.k8s.io/snapshot-promoter
+```
+Such Storage Class is necessary for restoring a Persistent Volume from already created Volume Snapshot and Volume Snapshot Data.
+
+## Creating
+Each `VolumeSnapshot` contains a spec and status, which is the specification and status of the Volume Snapshot.
+```yaml
+apiVersion: volume-snapshot-data.external-storage.k8s.io/v1
+kind: VolumeSnapshot
+metadata:
+  name: snapshot-demo
+spec:
+  persistentVolumeClaimName: ebs-pvc
+```
+
+* `persistentVolumeClaimName`: name of the Persistent Volume Claim that is bound to a Persistent Volume. This particular Persistent Volume will be snapshotted.
+
+Volume Snapshot Data is automatically created based on the Volume Snapshot. Relationship between Volume Snapshot and Volume Snapshot Data is similar to the relationship between Persistent Volume Claim and Persistent Volume.
+
+Depending on the Persistent Volume type the operation might go through several phases which are reflected by the Volume Snapshot status:
+1. The Volume Snapshot Data is created and is bound to the Volume Snapshot.
+2. The controller starts the snapshot operation: the snapshotted Persistent Volume might need to be frozen and the applications paused.
+3. The storage system finishes creating the snapshot (the snapshot is "cut") and the snapshotted Persistent Volume might return to normal operation. The snapshot itself is not ready yet. The last status condition is of `Created` type with status value "True"
+4. The newly created snapshot is completed and ready to use. The last status condition is of `Ready` type with status value "True"
+
+A Volume Snapshot status can be displayed as shown below:
+```yaml
+$ kubectl get volumesnapshot -o yaml
+apiVersion: volume-snapshot-data.external-storage.k8s.io/v1
+  kind: VolumeSnapshot
+  metadata:
+    clusterName: ""
+    creationTimestamp: 2017-09-19T13:58:28Z
+    generation: 0
+    labels:
+      Timestamp: "1505829508178510973"
+    name: snapshot-demo
+    namespace: default
+    resourceVersion: "780"
+    selfLink: /apis/volume-snapshot-data.external-storage.k8s.io/v1/namespaces/default/volumesnapshots/snapshot-demo
+    uid: 9cc5da57-9d42-11e7-9b25-90b11c132b3f
+  spec:
+    persistentVolumeClaimName: pvc-hostpath
+    snapshotDataName: k8s-volume-snapshot-9cc8813e-9d42-11e7-8bed-90b11c132b3f
+  status:
+    conditions:
+    - lastTransitionTime: null
+      message: Snapshot created successfully
+      reason: ""
+      status: "True"
+      type: Ready
+    creationTimestamp: null
+```
+
+## Restoring
```

Comment:
dup of snapshot promoter section?

Reply:
I'm doing my best to look at the snaphotting from a user perspective. That's why I split it into Prerequisites and the actual features.
So, firstly, what user needs before being able to use the snapshotting features, i.e. the prerequisites (existing Storage Class pointing to the Snapshot Promoter is one of them).
Secondly, the features, i.e. creating/restoring/deleting a snapshot.

That's why the Snapshot Promoter subsection is a part of the Prerequisites section and Restoring is a separate section as Creating and Deleting.

Does it make sense?

B.T.W., admin guide, i.e. how to install "all the necessary parts", doesn't exist so far.